airflow/operators/python.py

```diff
@@ -547,14 +547,20 @@ def __init__(
     ):
         if (
             python_version
-            and str(python_version)[0] != str(sys.version_info.major)
+            and str(python_version[0]) != str(sys.version_info.major)
```

Comment:
To deprecate a type, you need to keep it backward compatible. This is a breaking change for int/float values, where we cannot subscript a numeric value. Actually there is no need for this change, the current code works for both numerical and sting versions.

Reply:
fixed typo. thank you